SerialPrograms/Source/PokemonBDSP/Inference/BoxSystem/PokemonBDSP_BoxGenderDetector.cpp

```diff
@@ -0,0 +1,102 @@
+#include "PokemonBDSP_BoxGenderDetector.h"
+#include <QImage>
+#include "CommonFramework/ImageTools/ImageStats.h"
+#include "CommonFramework/ImageTools/ImageBoxes.h"
+#include "CommonFramework/ImageTools/ImageFilter.h"
+#include "PokemonBDSP/Options/PokemonBDSP_EggHatchFilter.h"
+#include "CommonFramework/ImageMatch/ExactImageMatcher.h"
+#include "CommonFramework/Globals.h"
+#include "Kernels/Waterfill/Kernels_Waterfill.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+
+namespace PokemonAutomation{
+namespace NintendoSwitch{
+namespace PokemonBDSP{
+
+using namespace Kernels;
+using namespace Kernels::Waterfill;
+
+
+const ImageMatch::ExactImageMatcher& FEMALE_ICON(){
+    static ImageMatch::ExactImageMatcher matcher(QImage(RESOURCE_PATH() + "PokemonBDSP/F-Icon.png"));
+    return matcher;
+}
+
+const ImageMatch::ExactImageMatcher& MALE_ICON(){
+    static ImageMatch::ExactImageMatcher matcher(QImage(RESOURCE_PATH() + "PokemonBDSP/M-Icon.png"));
+    return matcher;
+}
+
+bool is_male(const ConstImageRef& image, const WaterfillObject& object){
+    size_t width = object.width();
+    size_t height = object.height();
+    if (width > 2 * height){
+        return false;
+    }
+    if (height > 2 * width){
+        return false;
+    }
+
+    ConstImageRef obj = extract_box_reference(image, object);
+//    static int c = 0;
+//    obj.save("obj-" + QString::number(c++) + ".png");
+//    image.save("image-" + QString::number(c++) + ".png");
+
+    double rmsd = MALE_ICON().rmsd(obj);
+    //logger.log("MALE RMSD:" + std::to_string(rmsd));
+    return rmsd <= 110;
+}
+
+bool is_female(const ConstImageRef& image, const WaterfillObject& object){
+    size_t width = object.width();
+    size_t height = object.height();
+    if (width > 2 * height){
+        return false;
+    }
+    if (height > 2 * width){
+        return false;
+    }
+
+    ConstImageRef obj = extract_box_reference(image, object);
+    double rmsd = FEMALE_ICON().rmsd(obj);
+    //logger.log("FEMALE RMSD:" + std::to_string(rmsd));
+    return rmsd <= 140;
+}
+
+
+EggHatchGenderFilter BoxGenderDetector::identify_gender(LoggerQt& logger, VideoOverlay& overlay,const QImage& frame)
+{
+    InferenceBoxScope gender_box(overlay, 0.733, 0.022, 0.204, 0.049, COLOR_BLUE);
+
+    QImage name_and_gender = extract_box_reference(frame, gender_box).to_qimage();
```

Comment:
Since you're not changing the image, you can save the reference instead to avoid copying it.

`ConstImageRef image = extract_box_reference(frame, gender_box);`

Reply:
changed!